src/component/ResponsiveContainer.tsx

```diff
@@ -83,7 +82,16 @@ export const ResponsiveContainer = forwardRef(
         containerHeight: containerRef.current.clientHeight,
       };
     }, []);
+    const getDataProps = (props: Partial<Props>) => {
+      const dataProps: Partial<Record<keyof Props, Props[keyof Props]>> = {};
+      Object.keys(rest).forEach((prop: keyof Props) => {
+        if (prop.match(/^data-/)) {
+          dataProps[prop] = props[prop];
+        }
+      });
 
+      return dataProps;
+    };
```

Comment:
nice! we can probably extract this elsewhere though. It will probably get used again and it isn't specific to this file. Can add a test for it elsewhere as well. Maybe one of the util files?

Reply:
I think that's a good idea! Will do.